platforms/core-configuration/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslPlugin.kt

```diff
@@ -59,4 +73,48 @@ abstract class KotlinDslPlugin : Plugin<Project> {
             )
         }
     }
+
+    private
+    fun Project.workaroundKotlinJavaVersionSupport() {
+        val targetJavaVersion = kotlinDslOptionsJvmTarget
+            .orElse(defaultToolchainJavaVersion)
+            .map(::toLastSupportedKotlinJavaVersion)
+        tasks.withType(KotlinCompile::class.java).configureEach { kotlinCompile ->
+            kotlinCompile.compilerOptions.jvmTarget.set(
```

Comment:
Should it be `.convention`?

Reply:
Correct!

We will revert the workaround anyway given 8.5 now already have Kotlin 1.9.20 embedded which supports Java 21 and now automatically caps the jvm target to the latest supported.

* https://github.com/gradle/gradle/pull/26931